python_ta/debug/snapshot_manager.py

```diff
@@ -0,0 +1,94 @@
+from __future__ import annotations
+
+import inspect
+import logging
+import os
+import re
+import sys
+import types
+from typing import Any, Iterable, Optional
+
+from python_ta.debug.snapshot import snapshot
+
+
+class SnapshotManager:
+    """
+    Class used to manage the snapshots taken during the execution of a program.
```

Comment:
Should mention this context manager takes a snapshot on every line of function it's a part of